walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,122 +15,97 @@
+    if(requiredBusyRanges.size() > 0) requiredBusyRangesMerged.push(requiredBusyRanges.get(0));
+    
+    for(TimeRange timeRange : requiredBusyRanges){
+      if(timeRange.overlaps(requiredBusyRangesMerged.peek())){
+        topRange = requiredBusyRangesMerged.pop();
+        requiredBusyRangesMerged.push(mergeTimeRanges(topRange, timeRange));
+      }else{
+        requiredBusyRangesMerged.push(timeRange);
       }
     }
+    
   
-    optionalBusyRanges.push(optionalImportantRanges.get(0));
-    for(TimeRange timeRange : optionalImportantRanges){
-      if(timeRange.overlaps(optionalBusyRanges.peek())){
-        popedRange = optionalBusyRanges.pop();
-        optionalBusyRanges.push(mergeTimeRanges(popedRange, timeRange));
+    if(allBusyRanges.size() > 0) allBusyRangesMerged.push(allBusyRanges.get(0));
+    for(TimeRange timeRange : allBusyRanges){
+      if(timeRange.overlaps(allBusyRangesMerged.peek())){
+        topRange = allBusyRangesMerged.pop();
+        allBusyRangesMerged.push(mergeTimeRanges(topRange, timeRange));
       }else{
-        optionalBusyRanges.push(timeRange);
+        allBusyRangesMerged.push(timeRange);
       }
     }
 
-    //looking for free spaces among merged intervals
     int duration;
     //dummy TimeRanges to provide limits to the entire space
-    busyRanges.add(0, new TimeRange(TimeRange.START_OF_DAY, TimeRange.START_OF_DAY));
-    busyRanges.add(new TimeRange(TimeRange.END_OF_DAY+1, TimeRange.END_OF_DAY+1));
-    optionalBusyRanges.add(0, new TimeRange(TimeRange.START_OF_DAY, TimeRange.START_OF_DAY));
-    optionalBusyRanges.add(new TimeRange(TimeRange.END_OF_DAY+1, TimeRange.END_OF_DAY+1));
+    requiredBusyRangesMerged.add(new TimeRange(TimeRange.END_OF_DAY+1, TimeRange.END_OF_DAY+1));
+    requiredBusyRangesMerged.add(0, new TimeRange(TimeRange.START_OF_DAY, TimeRange.START_OF_DAY));
+    allBusyRangesMerged.add(new TimeRange(TimeRange.END_OF_DAY+1, TimeRange.END_OF_DAY+1));
+    allBusyRangesMerged.add(0, new TimeRange(TimeRange.START_OF_DAY, TimeRange.START_OF_DAY));
+
     //this loop calculates size of free space between two subsequent intervals, and checks if size meets requirements
     //if so it adds this space to freeIntervals that stores possible place for requested event
-    for(int i = 0; i < busyRanges.size()-1; i++){
-      duration = busyRanges.get(i+1).start()-busyRanges.get(i).end();
-      if(duration > 0 && duration >= request.getDuration())
-        freeRanges.add(new TimeRange(busyRanges.get(i).end(), busyRanges.get(i+1).start()-busyRanges.get(i).end()));
-    }
-    for(int i = 0; i < optionalBusyRanges.size()-1; i++){
-      duration = optionalBusyRanges.get(i+1).start()-optionalBusyRanges.get(i).end();
+    ArrayList<TimeRange> freeRanges = new ArrayList<TimeRange>();
+    for(int i = 0; i < allBusyRangesMerged.size()-1; i++){
+      duration = allBusyRangesMerged.get(i+1).start()-allBusyRangesMerged.get(i).end();
       if(duration > 0 && duration >= request.getDuration())
```

Comment:
I think we should check or just assume `request.getDuration() > 0`, in which case we don't need to check `duration > 0` again here.